app/services/idv/agent.rb

```diff
@@ -5,69 +5,143 @@ def initialize(applicant)
+          document_capture_session.store_proofing_result(idv_result[:resolution_result])
 
-        nil
+          nil
+        end
       end
     end
 
     def proof_address(document_capture_session, trace_id:)
+      document_capture_session.create_proofing_session
       callback_url = Rails.application.routes.url_helpers.address_proof_result_url(
         document_capture_session.result_id,
       )
 
-      LambdaJobs::Runner.new(
-        job_class: Idv::Proofer.address_job_class,
-        args: { applicant_pii: @applicant, callback_url: callback_url, trace_id: trace_id },
-        in_process_config: {
-          lexisnexis_config: {
-            phone_finder_workflow: AppConfig.env.lexisnexis_phone_finder_workflow,
-            account_id: AppConfig.env.lexisnexis_account_id,
-            base_url: AppConfig.env.lexisnexis_base_url,
-            username: AppConfig.env.lexisnexis_username,
-            password: AppConfig.env.lexisnexis_password,
-            request_mode: AppConfig.env.lexisnexis_request_mode,
-            request_timeout: AppConfig.env.lexisnexis_timeout,
+      if FeatureManagement.use_ruby_workers?
+        encrypted_arguments = Encryption::Encryptors::SessionEncryptor.new.encrypt(
+          { applicant_pii: @applicant }.to_json,
+        )
+        AddressProofingJob.perform_later(
+          encrypted_arguments: encrypted_arguments,
+          callback_url: callback_url,
+          result_id: document_capture_session.result_id,
+          trace_id: trace_id,
+        )
+      else
+        LambdaJobs::Runner.new(
+          job_class: Idv::Proofer.address_job_class,
+          args: { applicant_pii: @applicant, callback_url: callback_url, trace_id: trace_id },
+          in_process_config: {
+            lexisnexis_config: {
+              phone_finder_workflow: AppConfig.env.lexisnexis_phone_finder_workflow,
+              account_id: AppConfig.env.lexisnexis_account_id,
+              base_url: AppConfig.env.lexisnexis_base_url,
+              username: AppConfig.env.lexisnexis_username,
+              password: AppConfig.env.lexisnexis_password,
+              request_mode: AppConfig.env.lexisnexis_request_mode,
+              request_timeout: AppConfig.env.lexisnexis_timeout,
+            },
           },
-        },
-      ).run do |idv_result|
-        document_capture_session.store_proofing_result(idv_result[:address_result])
+        ).run do |idv_result|
+          document_capture_session.store_proofing_result(idv_result[:address_result])
+
+          nil
+        end
+      end
+    end
+
+    def proof_document(document_capture_session, liveness_checking_enabled:, trace_id:)
+      callback_url = Rails.application.routes.url_helpers.document_proof_result_url(
+        result_id: document_capture_session.result_id,
+      )
+
+      if FeatureManagement.use_ruby_workers?
+        encrypted_arguments = Encryption::Encryptors::SessionEncryptor.new.encrypt(
+          @applicant.to_json,
+        )
```

Comment:
Maybe we just add `ApplicationJob.encrypt_payload(@applicant)` or something>?

Reply:
I'm ok with the duplication for now since it's limited to this one file